fix(input): don't always use controlled mode

remorses opened the pull request:
closes: #1395 

## 📝 Description

Passing "" to react-aria forces the input to be in controlled mode, where there ref value is ignored

## ⛳️ Current behavior (updates)

Input does not use ref value

## 🚀 New behavior

Input can be initialized with `ref.current.value = 'xx'`, which is what react-hook-form does

## 💣 Is this a breaking change (Yes/No):

No

## 📝 Additional Information

remorses commented:
With this PR that issue should be solved too, the problem that caused the warning in the console was these lines:
```
 props.value ?? undefined,
props.defaultValue ?? undefined,
```
If you pass undefined `react-aria` will think that the input is not controlled, if instead you pass `""` it will switch the input to controlled mode, and not print the warning, 

it's the responsibility of the user to pass `""`, not Nextui, always passing `""` will cause much worse bugs like #1395

jrgarciadev commented:
> With this PR that issue should be solved too, the problem that caused the warning in the console was these lines:
> 
> ```ts
> props.value ?? undefined,
> props.defaultValue ?? undefined,
> ```
> 
> If you pass undefined `react-aria` will think that the input is not controlled, if instead you pass `""` it will switch the input to controlled mode, and not print the warning,
> 
> it's the responsibility of the user to pass `""`, not Nextui, always passing `""` will cause much worse bugs like #1395

The "Warning: A component is changing an uncontrolled input to be controlled. ..." is still there with your changes, I figured out that I was passing the value to the `useTextField` hook but that's not necessary, the solution is to remove the `value: inputValue` prop, could you update the PR please?, and test your use case

remorses commented:
Ok i removed the value from `useTextField` and my use case still works

I added a test for my use case, it does the check 2 times because previously the value was being reset asynchronously, it would first go to "value" and then ""

jrgarciadev commented:
Ok thanks!, please add the changeset as well to be able to publish the changes
